templates/CRM/Custom/Page/CustomDataView.tpl

```diff
@@ -47,41 +47,40 @@
                   </a>
                 </div>
               {/if}
-              {foreach from=$cd_edit.fields item=element key=field_id}
+              {if !empty($cd_edit.fields)}
                 <table class="crm-info-panel">
-                  <tr>
-                    {if $element.options_per_line != 0}
+                  {foreach from=$cd_edit.fields item=element key=field_id}
+                    <tr>
                       <td class="label">{$element.field_title}</td>
                       <td class="html-adjust">
-                        {* sort by fails for option per line. Added a variable to iterate through the element array*}
-                        {foreach from=$element.field_value item=val}
-                          {$val}
-                          <br/>
-                        {/foreach}
-                      </td>
-                    {else}
-                      <td class="label">{$element.field_title}</td>
-                      {if $element.field_data_type == 'Money'}
-                        {if $element.field_type == 'Text'}
-                          <td class="html-adjust">{$element.data|crmMoney}</td>
+                        {if $element.options_per_line != 0}
```

Comment:
This (56-61) appears to be useless, since the custom field value is never an array when we're viewing, as far as I can tell. Not that you added this in, but maybe worth getting rid off while we're here.